src/vivarium_public_health/population/mortality.py

```diff
@@ -83,13 +101,69 @@ def _get_mortality_rate(self, builder: Builder) -> Pipeline:
             requires_columns=["age", "sex"],
         )
 
+    # noinspection PyMethodMayBeStatic
+    def _get_all_cause_mortality_rate(self, builder: Builder) -> LookupTable:
```

Comment:
Should type the returns as either `Union[LookupTable, Pipeline]` or `Callable[[pd.Index], Union[pd.Series, pd.DataFrame]]` to make the typing flexible in inheritance.

Reply:
I think vivarium should perhaps have a custom type that corresponds to something like this. It's complicated somewhat by the fact that those two type definitions you suggested aren't actually equivalent.